### ISSUE TITLE
Sort todo lists

### ISSUE THREAD

**Issue by dvor** [vimwiki/vimwiki#349]
It would be nice to have ability to have "todo" items always on the top of a list and "done" items automatically moved to the bottom. This way there would be no need to manually clean up todo list and history of old items will be always available.

Some examples (`✓` symbol means that this item will be toggled):

```
- [ ] One        - [ ] One
- [✓] Two    ->  - [ ] Three
- [ ] Three      - [X] Two
- [X] Four       - [X] Four


- [ ] One                      - [ ] One
- [o] Two                      - [ ] Three
    - [✓] some subtask         - [X] Two
    - [X] another subtask   ->     - [X] some subtask
- [ ] Three                        - [X] another subtask
- [X] Four                     - [X] Four
```

**Comment by dwainm** [comment]
This can be done with `:sort` command

Visually select the list and the run `:'<,'>sort`

The completed ones will jump to the bottom. If you want it the other way around run:
`:'<,'>sort!`

**Comment by dvor** [comment]
@dwainm sort is not working correctly with nested list. It sorts list alphabetically, which is not desired behavior. For example

```
- [X] One                 - [ ] Two A
- [ ] Two                 - [ ] Two B
    - [ ] Two A  ->   - [ ] Three
    - [ ] Two B       - [ ] Two
- [ ] Three           - [X] One
```

**Comment by Nudin** [comment]
I have this on my todo-list. But will probably do only after the big refactoring is done. If someone else want's to do you're welcome!

**Comment by JasonMFry** [comment]
I just started using vimwiki and was looking for this feature in the help pages. I'm hoping to move this back closer to the top of the todo list :D

**Comment by nicksergeant** [comment]
@JasonMFry same here! Just started using and wanted a way to re-order todo items. I've settled on this very hacky (and dependent on your TODO structure) approach:

```
nnoremap <leader>t ?#<space><cr>jV}k :sort!<cr>:let @/=''<cr>
```

(https://github.com/nicksergeant/dotfiles/blob/3674006bd33d4508eb6aaf84b230d9b503cbe5d0/vimrc#L550)

**Comment by tinmarino** [comment]
I think this is an easy task as just a working function must be added `VimwikiSortTasks`
Feel free to try a PR.
